src/server/async_disconnect_handler.cpp

```diff
@@ -92,11 +92,12 @@ async_disconnect_handler<T_CONN_HANDLER>::disconnect_loop ()
 	m_disconnect_queue.swap (disconnect_work_buffer);
       }
 
-      while (!disconnect_work_buffer.empty ())
-	{
-	  connection_handler_uptr_t &front = disconnect_work_buffer.front ();
-	  front.reset (nullptr);
```

Comment:
Do we really need to set it to nullptr explicitly?

Reply:
not really; but the new code - albeit more correct - is also more cryptic :)